torch/fx/symbolic_trace.py

```diff
@@ -53,13 +65,51 @@ class DefaultDelegate:
     def is_leaf_module(self, m):
         return m.__module__.startswith('torch.nn') and not isinstance(m, torch.nn.Sequential)
 
-    # A method to insert a graph node given target, args, kwargs, and name.
-    # This method can be overridden to do extra checking, validation, or
-    # modification of values used in node creation. For example, one might
-    # want to disallow in-place operations from being recorded.
-    def create_node(self, graph : Graph, kind : str, target : Union[str, Callable],
-                    args : Tuple[Any], kwargs : Dict[str, Any], name : Optional[str] = None) -> Node:
-        return graph._create_node(kind, target, args, kwargs, name)
+    def create_arg(self, a):
+        # aggregates
+        if isinstance(a, (tuple, list)):
+            return type(a)(self.create_arg(elem) for elem in a)
+        elif isinstance(a, dict):
+            r = {}
+            for k, v in a.items():
+                if not isinstance(k, str):
+                    raise NotImplementedError(f"dictionaries with non-string keys: {a}")
+                r[k] = self.create_arg(v)
+            return r
+        elif isinstance(a, slice):
+            return slice(self.create_arg(a.start), self.create_arg(a.stop), self.create_arg(a.step))
+
+        if isinstance(a, torch.nn.Parameter):
+            for n, p in self.root.named_parameters():
+                if a is p:
+                    return self.graph.get_param(n)
+            raise NameError('parameter is not a member of this module')
+        elif isinstance(a, Proxy):
+            # base case: we unwrap the Proxy object
+            return a.node
+        elif isinstance(a, (str, int, float, bool, torch.dtype, torch.Tensor)) or a is None:
+            return a
+
+        raise NotImplementedError(f"argument of type: {type(a)}")
+
+
+class DefaultDelegate(DelegateBase):
+    def __init__(self, root: torch.nn.Module, graph: Graph):
+        super().__init__(graph)
+        self.root = root
+
+    def create_arg(self, a):        
+        if isinstance(a, torch.nn.Parameter):
+            for n, p in self.root.named_parameters():
+                if a is p:
+                    return self.graph.get_param(n)
+            raise NameError('parameter is not a member of this module')
```

Comment:
What's going on here? Isn't this covered in the base implementation?

Reply:
Yeah, I refactored wrong in this revision. The fixed version is now more clear: the base delegate cannot handle parameters because it doesn't have a `root` module associated with it. It is used primarily when you are just using the proxy object to add to a graph outside of symbolic tracing, e.g. `t = Proxy(node); output = (t *t - t).node`.

Reply:
The advantage of the "only Node" approach before was that you didn't have to wrap nodes in proxy objects to add more compute with them. But the problems it introduces out-weigh making it slightly more difficult to do.